### PR TITLE
feat: character list window

### PR DESCRIPTION
**Pull request by majestyotbr**
characterlist

### CODE REVIEW

**Review by Costallat — APPROVED** [review]

**Review by elsongabriel — APPROVED** [review]

**Comment by luanluciano93** [comment]
@majestyotbr you can replace (if you want): `if condition == true then` per `if condition then`

File: modules/client_entergame/characterlist.lua [hunk 1]
```diff
@@ -313,11 +355,14 @@ function CharacterList.create(characters, account, otui)
 
     if account.subStatus == SubscriptionStatus.Free then
         accountStatusLabel:setText(('%s%s'):format(tr('Free Account'), status))
+        accountStatusIcon:setImageSource('/images/game/entergame/nopremium')
     elseif account.subStatus == SubscriptionStatus.Premium then
         if account.premDays == 0 or account.premDays == 65535 then
```
Comment (luanluciano93) [inline modules/client_entergame/characterlist.lua:367]:
```suggestion
        if account.premDays == 0 or account.premDays == 65535 then
            accountStatusLabel:setText(('%s%s'):format(tr('Gratis Premium Account'), status))
        else
            accountStatusLabel:setText(('%s%s'):format(tr('Premium Account (%s) days left', account.premDays), status))
        end
        accountStatusIcon:setImageSource('/images/game/entergame/premium')
```

File: modules/client_entergame/characterlist.lua [hunk 2]
```diff
@@ -313,12 +355,14 @@ function CharacterList.create(characters, account, otui)
...
             accountStatusLabel:setText(('%s%s'):format(tr('Gratis Premium Account'), status))
         else
             accountStatusLabel:setText(('%s%s'):format(tr('Premium Account (%s) days left', account.premDays), status))
         end
+            accountStatusIcon:setImageSource('/images/game/entergame/premium')
```
Comment (luanluciano93) [inline modules/client_entergame/characterlist.lua:373]:
```suggestion
        accountStatusIcon:setImageSource('/images/game/entergame/premium')
```

**Review by lgrossi — APPROVED** [review]

**Review by luanluciano93 — APPROVED** [review]

**Comment by rodrigopaixaorj** [comment]
It should work with all client versions, not working with 860

image

**Comment by marcosvf132** [comment]
> It should work with all client versions, not working with 860
> 
> image

Not, it shouldnt. The structure of the login on 12+ client is completely different from the 11- clients.
The QT client (12+) get the characterlist and initial messages throught HTTP POST, using the login.php from your AAC.
The old client (11-) get the characterlist from the server protocollogin.

If you want this feature to work on all clients version perfectly you must change your protocollogin structure to include characters appearances.

By default the only data that comes from the server protocollogin is the character name, server name, server IP, game Port and premium days.

**Comment by Vysius** [comment]
> > It should work with all client versions, not working with 860
> > image
> 
> Not, it shouldnt. The structure of the login on 12+ client is completely different from the 11- clients. The QT client (12+) get the characterlist and initial messages throught HTTP POST, using the login.php from your AAC. The old client (11-) get the characterlist from the server protocollogin.
> 
> If you want this feature to work on all clients version perfectly you must change your protocollogin structure to include characters appearances.
> 
> By default the only data that comes from the server protocollogin is the character name, server name, server IP, game Port and premium days.

Then the PR should handle that, and keep the same aspect as old client, and this new only for newer protocols

**Comment by majestyotbr** [comment]
@marcosvf132 Thanks for the answer. How can i disable the old versions? Can you review then i change the code, thanks.

**Comment by majestyotbr** [comment]
@marcosvf132 Thanks for the answer. 
How can i check the old versions?

**Comment by rodrigopaixaorj** [comment]
> > It should work with all client versions, not working with 860
> > image
> 
> Not, it shouldnt. The structure of the login on 12+ client is completely different from the 11- clients. The QT client (12+) get the characterlist and initial messages throught HTTP POST, using the login.php from your AAC. The old client (11-) get the characterlist from the server protocollogin.
> 
> If you want this feature to work on all clients version perfectly you must change your protocollogin structure to include characters appearances.
> 
> By default the only data that comes from the server protocollogin is the character name, server name, server IP, game Port and premium days.

It should, atleast, must have a condition to check if version is higher than X, then use new character list, if lower, use old character list layout, since the main focus of OTClient is a compatibility with old and new tibia versions ;)

@majestyotbr Please, don't delete comments from users (Vysius) ;)

**Comment by majestyotbr** [comment]
> > It should, atleast, must have a condition to check if version is higher than X, then use new character list, if lower, use old character list layout, since the main focus of OTClient is a compatibility with old and new tibia versions ;)

That's why i asked for @marcosvf132 help.

**Comment by rodrigopaixaorj** [comment]
I was reading some files, and well, you can use **g_game.getClientVersion()**, so now, it's kinda easy to make it work

**Comment by majestyotbr** [comment]
> I was reading some files, and well, you can use **g_game.getClientVersion()**, so now, it's kinda easy to make it work

Thanks, i found **g_game.getClientVersion()**. This is exactly i need to do.

**Comment by majestyotbr** [comment]
> I was reading some files, and well, you can use **g_game.getClientVersion()**, so now, it's kinda easy to make it work

Thanks, i found **g_game.getClientVersion()**. This is exactly what i need to do.

**Comment by majestyotbr** [comment]
@Nottinghster 
@marcosvf132 helped with the old protocol compatibily, can you test? Thanks.

**Comment by majestyotbr** [comment]
@Nottinghster 
@marcosvf132 helped with the old protocol compatibility, can you test? Thanks.

**Comment by rodrigopaixaorj** [comment]
> @Nottinghster @marcosvf132 helped with the old protocol compatibility, can you test? Thanks.

Sure, later I'll test, can't atm, on job :)

**Review by mehah — APPROVED** [review]

**Pull request merged**